src/commands.js

```diff
@@ -0,0 +1,566 @@
+    }
+  }
+
+  selectNextResult(forward) {
+    const all = this.renderRoot.querySelectorAll("a.hit");
+    let selected;
+    let selectedIndex;
+    for (const [index, element] of all.entries()) {
+      if (element.classList.contains("active")) {
+        selected = element;
+        selectedIndex = index;
+        break;
+      }
+    }
+
+    const lastIndex = all.length > 0 ? all.length - 1 : 0;
+
+    let nextIndex = 0;
+    if (selectedIndex !== undefined) {
+      nextIndex = forward ? selectedIndex + 1 : selectedIndex - 1;
+    }
+
+    // Check if we're at the end/start of the list, and adjust accordingly
+    if (nextIndex > lastIndex) {
+      nextIndex = 0;
+    } else if (nextIndex < 0) {
+      nextIndex = lastIndex;
+    }
+
+    // Remove all active elements
+    for (const active of this.renderRoot.querySelectorAll("a.hit.active")) {
+      active.classList.remove("active");
+    }
+
+    // Add class for active element and scroll to it
+    const newActive = all[nextIndex];
+    newActive.classList.add("active");
+    newActive.scrollIntoView({
+      behavior: "smooth",
+      block: "nearest",
+      inline: "start",
+    });
+  }
+
+  mouseenterResultHit(e) {
+    console.log("Mouse enter result hit");
+    const activeElements = this.renderRoot.querySelectorAll("a.hit.active");
+    for (const element of activeElements) {
+      element.classList.remove("active");
+    }
+    e.currentTarget.classList.add("active");
+  }
+
+  triggerClosePalette() {
+    console.log("Triggering close palette");
+    const event = new CustomEvent(EVENT_READTHEDOCS_COMMANDS_HIDE);
+    document.dispatchEvent(event);
+  }
+
+  showPalette() {
+    console.log("Showing palette");
+    this.show = true;
+    this.updateComplete.then(() => {
+      const input = this.renderRoot.querySelector("input[type=search]");
+      if (input) {
+        input.focus();
+      }
+    });
+  }
+
+  closePalette() {
+    console.log("Closing palette");
+    this.show = false;
+  }
+
+  _handleClosePalette = (e) => {
+    e.preventDefault();
+    this.closePalette();
+  };
+
+  _handleShowPalette = (e) => {
+    e.preventDefault();
+    this.showPalette();
+  };
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    document.addEventListener(
+      EVENT_READTHEDOCS_COMMANDS_SHOW,
+      this._handleShowPalette,
+    );
+    document.addEventListener(
+      EVENT_READTHEDOCS_COMMANDS_HIDE,
+      this._handleClosePalette,
+    );
+  }
+
+  disconnectedCallback() {
+    document.removeEventListener(
+      EVENT_READTHEDOCS_COMMANDS_SHOW,
+      this._handleShowPalette,
+    );
+    document.removeEventListener(
+      EVENT_READTHEDOCS_COMMANDS_HIDE,
+      this._handleClosePalette,
+    );
+
+    super.disconnectedCallback();
+  }
+}
+
+export class CommandsAddon extends AddonBase {
+  static jsonValidationURI =
+    "http://v1.schemas.readthedocs.org/addons.commands.json";
+  static addonEnabledPath = "addons.filesections.enabled";
```

Comment:
The API should return `addons.commands`. File sections seems to be one of the features of the commands palette, not the addon itself.

```suggestion
  static addonEnabledPath = "addons.commands.enabled";
```